From c20aa4fb5f1d9f6506703b5587c9f6e4ca573cec Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Tue, 4 Feb 2020 09:42:15 +0530
Subject: [PATCH] Fix-APL_NUC-boot-issues-without-display-connection

When the period is updated due to a config change,
cache the period and don't immediately update the model until we actually observe updated vsyncs from the hardware.
But if never received the first resync sample, update the mPeriod immediately.

Tracked-On: https://jira.devtools.intel.com/browse/OAM-81129
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 services/surfaceflinger/Scheduler/DispSync.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/surfaceflinger/Scheduler/DispSync.cpp b/services/surfaceflinger/Scheduler/DispSync.cpp
index 0c9405297..72c49559b 100644
--- a/services/surfaceflinger/Scheduler/DispSync.cpp
+++ b/services/surfaceflinger/Scheduler/DispSync.cpp
@@ -648,6 +648,14 @@ void DispSync::setPeriod(nsecs_t period) {
 
     if (pendingPeriodShouldChange) {
         mPendingPeriod = period;
+		// On this time, this is first get the period config, set it immediately update.
+		// Don't wait the hardware vsync.
+        if (mNumResyncSamples == 0) {
+			 ALOGD("First resync sample doesn't received, update the mPeriod immediately");
+			 mPeriod = period;
+			 mPhase = 0;
+			 mThread->updateModel(mPeriod, mPhase, mReferenceTime);
+        }
     }
     if (mTraceDetailedInfo) {
         ATRACE_INT("DispSync:IntendedPeriod", mIntendedPeriod);
-- 
2.17.1

